packages/tenderly-core/src/internal/core/services/TenderlyApiService.ts

```diff
@@ -1,35 +1,59 @@
 import * as axios from "axios";
 
+import { logger } from "../../../utils/logger";
 import { getConfig, isAccessTokenSet } from "../../../utils/config";
 import { TENDERLY_API_BASE_URL, TENDERLY_JSON_RPC_BASE_URL } from "../../../common/constants";
 
 export class TenderlyApiService {
   public static configureInstance(): axios.AxiosInstance {
+    logger.debug("Configuring instance...");
+
     const tdlyConfig = getConfig();
-    return axios.default.create({
+    const params = {
+      baseURL: TENDERLY_API_BASE_URL,
+      accessKey: tdlyConfig.access_key,
+    };
+
+    logger.debug("Configured instance with parameters:", {
       baseURL: TENDERLY_API_BASE_URL,
-      headers: { "x-access-key": tdlyConfig.access_key },
+      access_key:
+        tdlyConfig.access_key !== undefined && tdlyConfig.access_key !== null && tdlyConfig.access_key !== ""
+          ? "super secret access_key is set in 'access_key' field"
+          : "undefined or null or empty string",
     });
+
+    return axios.default.create(params);
   }
 
   public static configureAnonymousInstance(): axios.AxiosInstance {
+    logger.debug("Configured anonymous instance with base url:", TENDERLY_API_BASE_URL);
+
     return axios.default.create({
       baseURL: TENDERLY_API_BASE_URL,
     });
   }
 
   public static configureTenderlyRPCInstance(): axios.AxiosInstance {
+    logger.debug("Configuring tenderly RPC instance...");
+
     const tdlyConfig = getConfig();
-    return axios.default.create({
+    const params = {
       baseURL: TENDERLY_JSON_RPC_BASE_URL,
       headers: {
         "x-access-key": tdlyConfig.access_key,
         Head: tdlyConfig.head !== undefined ? tdlyConfig.head : "",
       },
-    });
+    };
+    logger.debug("Configured tenderly rpc instance with parameters:", params);
+
+    return axios.default.create(params);
   }
 
   public static isAuthenticated(): boolean {
-    return isAccessTokenSet();
+    logger.debug("Getting if user is authenticated...");
+    const isAuth = isAccessTokenSet();
+    logger.debug("isAuthenticated returned:", isAuth);
```

Comment:
This is not a usable log message, it needs to be better.
Put yourself in a position where you read this after one year if it will be helpful for you.